### PR TITLE
Use TrajectoryNode::Data in FastCorrelativeMatcher.

### PR DESCRIPTION
**Pull request by cschuet**
PAIR=wohe@

### CODE REVIEW

File: cartographer/mapping_3d/sparse_pose_graph/constraint_builder.cc [hunk 1]
```diff
@@ -273,8 +265,9 @@ void ConstraintBuilder::FinishComputation(const int computation_index) {
     if (pending_computations_.empty()) {
       CHECK_EQ(submap_queued_work_items_.size(), 0);
       if (when_done_ != nullptr) {
-        for (const std::unique_ptr<OptimizationProblem::Constraint>&
-                 constraint : constraints_) {
+        for (
```
Comment (wohe) [inline cartographer/mapping_3d/sparse_pose_graph/constraint_builder.cc]: Reformat?

File: cartographer/mapping_3d/scan_matching/fast_correlative_scan_matcher_test.cc [hunk 2]
```diff
@@ -158,20 +172,21 @@ TEST_F(FastCorrelativeScanMatcherTest, CorrectPoseForMatchFullSubmap) {
...
-      Eigen::Quaterniond::Identity(), point_cloud_, point_cloud_, kMinScore,
-      [](const transform::Rigid3f&) { return kPassingLowResolutionScore; },
-      &score, &pose_estimate, &rotational_score, &low_resolution_score));
+      Eigen::Quaterniond::Identity(), CreateConstantData(point_cloud_),
+      kMinScore, &score, &pose_estimate, &rotational_score,
+      &low_resolution_score));
   EXPECT_LT(kMinScore, score);
   EXPECT_LT(0.09f, rotational_score);
-  EXPECT_LT(0.99f, low_resolution_score);
+  EXPECT_LT(0.09f, low_resolution_score);
```
Comment (wohe) [inline cartographer/mapping_3d/scan_matching/fast_correlative_scan_matcher_test.cc]: `0.14f`? Same above.

File: cartographer/mapping_3d/sparse_pose_graph/constraint_builder.cc (hunk 1 above)
Comment (cschuet) [reply cartographer/mapping_3d/sparse_pose_graph/constraint_builder.cc]: Done

File: cartographer/mapping_3d/scan_matching/fast_correlative_scan_matcher_test.cc (hunk 2 above)
Comment (cschuet) [reply cartographer/mapping_3d/scan_matching/fast_correlative_scan_matcher_test.cc]: Done

**Pull request merged**